### PR TITLE
Run yast2 scc by launch_yast2_module_x11 in SCC addon test poo#25920

### PR DESCRIPTION
**Pull request by thehejik**
Fix for addon_products_via_SCC_yast2 to get it working again. Mainly due changes in x11_start_program function.  We don't need check if yast is running by needle.

Tested here: http://dhcp62.suse.cz/tests/2593

### CODE REVIEW

File: tests/installation/addon_products_via_SCC_yast2.pm [hunk 1]
```diff
@@ -28,7 +28,7 @@ sub run {
         script_sudo "echo \"@addon_proxy.proxy.scc.suse.de\" > /etc/SUSEConnect", 0;
     }
     type_string "killall xterm\n";
-    x11_start_program("xdg-su -c '/sbin/yast2'", target_match => 'yast2-control-center-ui');
+    x11_start_program("xdg-su -c '/sbin/yast2'", valid => 0 );
```
Comment (rwx788) [inline tests/installation/addon_products_via_SCC_yast2.pm]: Please, consider launch_yast2_module_x11 function from y2x11test: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/y2x11test.pm#L16
It's a wrapper for yast modules, which includes password typing logic. But packagekit-warning logic is not in there, but it doesn't hurt to move it there.

**Comment by okurz** [comment]
see what @rwx788 suggested. Also see https://progress.opensuse.org/issues/25920. I think you are solving that ticket :)

**Comment by thehejik** [comment]
Ok, I made it similar to addon_products_yast2.pm using launch_yast2_module_x11

SLE12SP3 http://dhcp62.suse.cz/tests/2610
SLE15 http://dhcp62.suse.cz/tests/2612

... actually the test is failing for SLE15 but the reason is different than before (SLE15 repository related). Currently SLE15 is out of my scope and probably someone who knows SLE15 can identify/fix the problem better than me. I also expect that some needles for SLE15 will be missing.

I need the test in production for upcoming SLE12 SP3 GM SLERT

**Comment by okurz** [comment]
merged. Yes, the SLE15 problem looks different. The URL includes the proxy-SCC host name. I wonder what is expected here but to see that problem on osd in production after your fix can also help then to provide a proper bug report. Thanks

**Pull request merged**